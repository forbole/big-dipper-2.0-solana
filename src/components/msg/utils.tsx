import * as MODELS from '@models';
import * as R from 'ramda';
import {
  Tag,
} from '@components';
import * as COMPONENTS from '@msg';

const getDataByType = (type: string) => {
  // =====================================
  // DO NOT UPDATE IF THIS IS A FORK.
  // ONLY COSMOS SDK DEFAULT MESSAGES HERE.
  // Please use `customTypeToModel` below for custom message types
  // =====================================
  const defaultTypeToModel = {
    // ========================
    // staking
    // ========================
    // '/cosmos.staking.v1beta1.MsgDelegate': {
    //   model: MODELS.MsgDelegate,
    //   content: COMPONENTS.Delegate,
    //   tagTheme: 'one',
    //   tagDisplay: 'txDelegateLabel',
    // },
  };

  // =====================================
  // Update your chain's message types here
  // =====================================
  const customTypeToModel = {

  };

  if (defaultTypeToModel[type]) return defaultTypeToModel[type];
  if (customTypeToModel[type]) return customTypeToModel[type];
  return null;
};

/**
 * Helper function that helps get model by type
 * @param type Model type
 */
export const getMessageModelByType = (type: string) => {
  const data = getDataByType(type);
  if (data) {
    return data.model;
  }

  return MODELS.MsgUnknown;
};

/**
 * Helper function to correctly display the correct UI
 * @param type Model type
 */
export const getMessageByType = (message: any, viewRaw: boolean, t:any) => {
  const { type } = message;
  let results: {
    content: any;
    tagDisplay: string;
    tagTheme?: TagTheme;
    unknown?: boolean;
  } = {
    content: COMPONENTS.Unknown,
    tagDisplay: 'txUnknownLabel',
    tagTheme: 'zero',
  };

  const data = getDataByType(type);

  if (data) {
    results = {
      content: data?.content,
      tagDisplay: data.tagDisplay,
      tagTheme: data.tagTheme,
    };
  }

  // If user asks to view the raw data
  if (viewRaw || !results.content) {
    results.content = COMPONENTS.Unknown;
  }

  return {
    type: <Tag
      value={t(`message_labels:${results.tagDisplay}`)}
      theme={results.tagTheme}
    />,
    message: <results.content message={message as any} />,
  };
};

export const convertMsgsToModels = (transaction: any) => {
  const messages = R.pathOr([], ['messages'], transaction).map((msg, i) => {
    const model = getMessageModelByType(msg?.['@type']);
    if (model === MODELS.MsgWithdrawDelegatorReward
      || model === MODELS.MsgWithdrawValidatorCommission) {
      const log = R.pathOr(null, ['logs', i], transaction);
      return model.fromJson(msg, log);
    }
    return model.fromJson(msg);
  });

  return messages;
};
