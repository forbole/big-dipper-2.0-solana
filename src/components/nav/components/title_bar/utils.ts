export const formatMarket = (data: {
  marketCap: number;
  maxSupply: number;
  price: number;
  inflation: number;
}) => {
  return ([
    {
      key: 'marketCap',
      data: data.marketCap,
    },
    {
      key: 'maxSupply',
      data: data.maxSupply,
    },
    {
      key: 'price',
      data: data.price,
    },
    {
      key: 'inflation',
      data: data.inflation,
    },
  ]);
};
