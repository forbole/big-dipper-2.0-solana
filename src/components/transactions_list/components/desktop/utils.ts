export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'slot',
    width: 15,
  },
  {
    key: 'hash',
    width: 30,
  },
  {
    key: 'instructions',
    align: 'right',
    width: 15,
  },
  {
    key: 'result',
    align: 'right',
    width: 20,
  },
  {
    key: 'time',
    align: 'right',
    width: 20,
  },
];
