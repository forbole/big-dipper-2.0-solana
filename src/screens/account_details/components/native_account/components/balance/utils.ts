import { formatNumber } from '@utils/format_token';
import { BalanceType } from './types';

export const formatBalanceData = (data: BalanceType) => {
  const balanceChart = [
    {
      key: 'balanceNative',
      display: `${formatNumber(data.native.value, data.native.exponent)} ${data.native.displayDenom.toUpperCase()}`,
      value: data.native.value,
    },
    {
      key: 'balanceStake',
      display: `${formatNumber(data.stake.value, data.stake.exponent)} ${data.stake.displayDenom.toUpperCase()}`,
      value: data.stake.value,
    },
    {
      key: 'balanceNonce',
      display: `${formatNumber(data.nonce.value, data.nonce.exponent)} ${data.nonce.displayDenom.toUpperCase()}`,
      value: data.nonce.value,
    },
    {
      key: 'balanceVote',
      display: `${formatNumber(data.vote.value, data.vote.exponent)} ${data.vote.displayDenom.toUpperCase()}`,
      value: data.vote.value,
    },
  ];

  return balanceChart;
};
