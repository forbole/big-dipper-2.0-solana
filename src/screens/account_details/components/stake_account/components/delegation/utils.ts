import numeral from 'numeral';
import { DelegationType } from './types';

export const formatBalanceData = (data: DelegationType) => {
  const balanceChart = [
    {
      key: 'balanceActivated',
      display: `${numeral(data.activated.value).format(data.activated.format)} ${data.activated.denom.toUpperCase()}`,
      value: data.activated.value,
    },
    {
      key: 'balanceActivating',
      display: `${numeral(data.activating.value).format(data.activating.format)} ${data.activating.denom.toUpperCase()}`,
      value: data.activating.value,
    },
  ];

  return balanceChart;
};
