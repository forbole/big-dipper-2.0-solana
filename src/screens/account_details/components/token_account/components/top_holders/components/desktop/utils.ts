export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'rank',
    width: 5,
    align: 'center',
  },
  {
    key: 'holder',
    width: 65,
  },
  {
    key: 'amount',
    width: 30,
    align: 'right',
  },
];
