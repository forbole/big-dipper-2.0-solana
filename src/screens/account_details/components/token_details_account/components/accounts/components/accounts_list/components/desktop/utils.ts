export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'address',
    width: 95,
  },
];
