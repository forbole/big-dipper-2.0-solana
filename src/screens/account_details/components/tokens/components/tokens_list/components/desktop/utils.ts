export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'token',
    width: 50,
  },
  {
    key: 'amount',
    align: 'right',
    width: 50,
  },
];
