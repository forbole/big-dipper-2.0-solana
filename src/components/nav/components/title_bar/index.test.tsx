import React from 'react';
import { RecoilRoot } from 'recoil';
import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import TitleBar from '.';

// ==================================
// mocks
// ==================================
jest.mock('recoil', () => ({
  ...jest.requireActual('recoil') as any,
  useRecoilValue: jest.fn(() => ({
    marketCap: 66851913386,
    price: 218.98,
    maxSupply: {
      value: '11432342352345',
      exponent: 8,
      baseDenom: 'sol',
      displayDenom: 'sol',
    },
    inflation: 0.14,
  })),
}));

// ==================================
// unit tests
// ==================================
describe('screen: Nav/TitleBar', () => {
  it('hook toggles correctly', () => {
    const component = renderer.create(
      <RecoilRoot>
        <MockTheme>
          <TitleBar title="hello world" />
        </MockTheme>
      </RecoilRoot>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const pList = component.root.findAllByType('p');
    expect(pList.some((el) => el.children.includes('14%'))).toBe(true);
    expect(pList.some((el) => el.children.includes('0.13'))).toBe(false);
    expect(pList.some((el) => el.children.includes('$66,851,913,386'))).toBe(true);
    expect(pList.some((el) => el.children.includes('common:maxSupply'))).toBe(true);
  });
});

afterEach(() => {
  jest.clearAllMocks();
});
