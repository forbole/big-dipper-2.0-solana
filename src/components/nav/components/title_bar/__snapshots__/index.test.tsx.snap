// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Nav/TitleBar hook toggles correctly 1`] = `
<div
  className="makeStyles-root"
>
  <h1
    className="MuiTypography-root MuiTypography-h1"
  >
    hello world
  </h1>
  <div
    className="makeStyles-content"
  >
    <div
      className="makeStyles-item"
    >
      <p
        className="MuiTypography-root label MuiTypography-body1"
      >
        common:marketCap
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        $66,851,913,386
      </p>
    </div>
    <div
      className="makeStyles-item"
    >
      <p
        className="MuiTypography-root label MuiTypography-body1"
      >
        common:maxSupply
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        11,432,342,352,345 SOL
      </p>
    </div>
    <div
      className="makeStyles-item"
    >
      <p
        className="MuiTypography-root label MuiTypography-body1"
      >
        common:price
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        $218.98
      </p>
    </div>
    <div
      className="makeStyles-item"
    >
      <p
        className="MuiTypography-root label MuiTypography-body1"
      >
        common:inflation
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        14%
      </p>
    </div>
  </div>
</div>
`;
